src/_pytest/config/__init__.py

```diff
@@ -42,7 +42,11 @@
 from _pytest.pathlib import Path
 from _pytest.warning_types import PytestConfigWarning
 
-if False:  # TYPE_CHECKING
+try:
```

Comment:
I'm OK with the overall idea, but we should move this boilerplate to `_pytest.compat`:

```python
from _pytest.compat import TYPE_CHECKING

if TYPE_CHECKING:
    ...
```

Reply:
Oh also, I prefer we check for the specific versions instead of rely on an `ImportError`:

```python

if sys.version_info < (3, 5, 2):
    from typing_extensions import TYPE_CHECKING
else:
    from typing import TYPE_CHECKING
```